obp-api/src/main/scala/code/bankconnectors/LocalMappedConnector.scala

```diff
@@ -1316,19 +1293,50 @@ object LocalMappedConnector extends Connector with MdcLoggable {
                                   accountType: String,
                                   accountLabel: String,
                                   branchId: String,
-                                  accountRoutingScheme: String,
-                                  accountRoutingAddress: String,
+                                  accountRoutings: List[AccountRouting],
                                   callContext: Option[CallContext]
                                 ): OBPReturnType[Box[BankAccount]] = Future {
+
+    val oldAccountRoutings = BankAccountRouting.findAll(By(BankAccountRouting.BankId, bankId.value),
+      By(BankAccountRouting.AccountId, accountId.value))
+      .map(_.accountRouting)
+
+    // Add or update new routing schemes
+    accountRoutings.foreach(accountRouting =>
+      oldAccountRoutings.find(_.scheme == accountRouting.scheme) match {
+        case Some(_) =>
+          BankAccountRouting
+            .find(By(BankAccountRouting.BankId, bankId.value), By(BankAccountRouting.AccountId, accountId.value),
+              By(BankAccountRouting.AccountRoutingScheme, accountRouting.scheme))
+            .map(_.AccountRoutingAddress(accountRouting.address).saveMe())
+        case None =>
+          BankAccountRouting.create
+            .BankId(bankId.value)
+            .AccountId(accountId.value)
+            .AccountRoutingScheme(accountRouting.scheme)
+            .AccountRoutingAddress(accountRouting.address)
+            .saveMe()
+      }
+    )
+
+    // Delete non-present routing schemes
+    oldAccountRoutings.foreach(accountRouting =>
+      accountRoutings.find(_.scheme == accountRouting.scheme)
+        .getOrElse(
+          BankAccountRouting
+            .find(By(BankAccountRouting.BankId, bankId.value), By(BankAccountRouting.AccountId, accountId.value),
+              By(BankAccountRouting.AccountRoutingScheme, accountRouting.scheme))
+            .map(_.delete_!)
+        )
+    )
+
```

Comment:
The whole method body should change to:
```
   val oldAccountRoutings: List[BankAccountRouting] = BankAccountRouting.findAll(By(BankAccountRouting.BankId, bankId.value),
      By(BankAccountRouting.AccountId, accountId.value))
    
    // Add or update new routing schemes
    accountRoutings.foreach(accountRouting =>
      oldAccountRoutings.find(_.AccountRoutingScheme.get == accountRouting.scheme) match {
        case Some(v) =>
          v.AccountRoutingAddress(accountRouting.address).saveMe()
        case None =>
          BankAccountRouting.create
            .BankId(bankId.value)
            .AccountId(accountId.value)
            .AccountRoutingScheme(accountRouting.scheme)
            .AccountRoutingAddress(accountRouting.address)
            .saveMe()
      }
    )

    // Delete non-present routing schemes
    oldAccountRoutings.filterNot(accountRouting => accountRoutings.exists(_.scheme == accountRouting.AccountRoutingScheme.get))
      .foreach(_.delete_!)

    (for {
      (account, _) <- LocalMappedConnector.getBankAccountCommon(bankId, accountId, callContext)
    } yield {
      account
        .kind(accountType)
        .accountLabel(accountLabel)
        .mBranchId(branchId)
        .saveMe
    }, callContext)
```

Reply:
I took the main of your proposition, just for `_.AccountRoutingScheme.get` I replaced it by `_.accountRouting.scheme`. I think it's better and don't change anything.